synapse/handlers/appservice.py

```diff
@@ -307,6 +314,23 @@ async def _notify_interested_services_ephemeral(
                             service, "presence", new_token
                         )
 
+                    elif (
+                        stream_key == "to_device_key"
+                        and self.msc2409_to_device_messages_enabled
+                    ):
+                        # Retrieve an iterable of to-device message events, as well as the
```

Comment:
an iterable is insufficient, given we need to check if it is non-empty.

also, `submit_ephemeral_events_for_as` claims to want a list rather than an iterable (though looking at it, I think the annotation is overly specific)